omnipaxos/src/storage.rs

```diff
@@ -133,8 +127,11 @@ pub trait Storage<T>
 where
     T: Entry,
 {
-    /// Atomically perform and persist all storage operations in order.
-    /// Returns an Error if the transaction was not committed (so it was rolled back).
+    /// **Atomically** perform all storage operations in order.
+    /// For correctness, the operations must be atomic i.e., either all operations are performed
+    /// successfully or all get rolled back. If the `StorageResult` returns as `Err`, the
+    /// operations are assumed to have been rolled back to the previous state before this function
+    /// call.
     fn perform_ops_atomically(&mut self, batch: Vec<StorageOp<T>>) -> StorageResult<()>;
```

Comment:
Since we don't do any reads, perhaps it would be even more clear for the users if we rename it:

```suggestion
    fn write_atomically(&mut self, ops: Vec<StorageOp<T>>) -> StorageResult<()>;
```